packages/react-core/src/Authenticator/hooks/useAuthenticator/utils.ts

```diff
@@ -0,0 +1,37 @@
+import { areEmptyArrays, areEmptyObjects } from '../../../utils';
+import { Comparator, Selector } from './types';
+
+/**
+ * Does an ordering and shallow comparison of each array value,
+ * plus a value equality check for empty objects and arrays.
+ */
+export function areSelectorDepsEqual<T>(
+  currentDeps: T[],
+  nextDeps: T[]
+): boolean {
+  if (currentDeps.length !== nextDeps.length) {
+    return false;
+  }
+  return currentDeps.every((currentDep, index) => {
+    const nextDep = nextDeps[index];
+
+    if (
+      areEmptyArrays(currentDep, nextDep) ||
+      areEmptyObjects(currentDep, nextDep)
+    ) {
+      return true;
+    }
+
+    return currentDep === nextDep;
```

Comment:
Could `currentDep` or `nextDep` be a non-empty array or object?

Reply:
Without doing a deep comparison, comparing the non-empty arrays or objects with the same contents will always return `false`